bounce/server/api/membership.py

```diff
@@ -18,42 +18,72 @@ class MembershipEndpoint(Endpoint):
 
     __uri__ = "/memberships/<club_name:string>"
 
+    @verify_token()
     @validate(GetMembershipRequest, GetMembershipResponse)
-    async def get(self, request, club_name):
+    async def get(self, request, club_name, id_from_token=None):
         """
-        Handles a GET /memberships/<club_name>?user_id=<user_id> request
+        Handles a GET /memberships/<club_name> request
         by returning the membership that associates the given user with the
         given club. If no user ID is given, returns all memberships for the
         given club.
         """
         # Decode the club name
         club_name = unquote(club_name)
-        user_id = request.args.get('user_id', None)
+        # Get the body from the JSON requests
+        body = util.strip_whitespace(request.json)
 
         # Make sure the club exists
         club_row = club.select(self.server.db_session, club_name)
         if not club_row:
             raise APIError('No such club', status=404)
-
-        # Fetch the club's memberships
-        membership_info = membership.select(
-            self.server.db_session, club_name, user_id=user_id)
+        try:
+            user_id=body.get('user_id', None),
+            membership_attr=membership.select(
+                self.server.db_session,
+                club_name,
+                id_from_token,
+                Roles.president                
+            )
+            editors_role=membership_attr.get('role')
+            # Fetch the club's memberships
+            membership_info = membership.select(
+                self.server.db_session, 
+                club_name,
+                user_id,
+                editors_role)
+
+        except PermissionError:
+            raise APIError('Unauthorized', status=403)
         return response.json(membership_info, status=200)
 
     # pylint: disable=unused-argument
     @verify_token()
     @validate(PutMembershipRequest, None)
     async def put(self, request, club_name, id_from_token=None):
-        """Handles a PUT /memberships/<club_name>?user_id=<user_id> request by
```

Comment:
Does getting from the params mean the same as getting it from the uri? Or is it just getting it from the params in the json schema? Also, for user id why do you suggest to retrieve from params rather than from the body of the json schema?